packages/sockets/src/logic/creators/createSocketCommunicator.ts

```diff
@@ -50,6 +50,25 @@ export default function createSocketCommunicator({
     }
   }
 
+  function clearBuffer(buf: Array<unknown>) {
+    buf.length = 0;
+  }
+
+  function areProcessesEqual(a: Process, b: Process) {
+    return a.rank === b.rank;
+  }
```

Comment:
todo(non-blocking): looks like useful util, which can be extracted into `utils` in the current module or in `connections`